api/src/models/user/user.types.ts

```diff
@@ -42,7 +42,7 @@ export interface IUserDocument extends IUser, Document {
   addConnectionToUser: (this: IUserDocument,  objId: string, isTeamMate?: boolean) => Promise<IUserDocument>;
   deleteConnectionFromUser: (this: IUserDocument,  objId: string) => Promise<IUserDocument>;
   updateUserProfile: (this: IUserDocument, profileData: IProfileData) => Promise<IUserDocument>;
-  createAndPostThread: (this: IUserDocument, threadDetails: IThreadPostDetails) => Promise<IUserDocument>;
+  createAndPostThread: (this: IUserDocument, threadDetails: IThreadPostDetails) => Promise<(IUserDocument | IThreadDocument)[]>;
```

Comment:
`IUserDocument | IThreadDocument)[]` this doesn't seem right...

Reply:
I'll update this to an object as per your comment above